test/cluster/object_store/test_backup.py

```diff
@@ -748,3 +748,72 @@ async def test_restore_with_non_existing_sstable(manager: ManagerClient, object_
     print(f'Status: {status}')
     assert 'state' in status and status['state'] == 'failed'
     assert 'error' in status and 'Not Found' in status['error']
+
+
+@pytest.mark.asyncio
+async def test_backup_broken_streaming(manager: ManagerClient, s3_storage):
+    # Define configuration for the servers.
+    objconf = s3_storage.create_endpoint_conf()
+    config = {
+        'enable_user_defined_functions': False,
+        'object_storage_endpoints': objconf,
+        'experimental_features': ['keyspace-storage-options'],
+        'task_ttl_in_seconds': 300,
+    }
+    cmd = ['--smp', '1', '--logger-log-level', 'sstables_loader=debug:sstable=debug']
+    servers = await manager.servers_add(servers_num=1, config=config, cmdline=cmd, auto_rack_dc="dc1")
+
+    # Obtain the CQL interface from the manager.
+    cql = manager.get_cql()
+
+    keyspace = 'test_ks'
+    table = 'test_cf'
+    replication_opts = format_tuples({
+        'class': 'NetworkTopologyStrategy',
+        'replication_factor': '1'
+    })
+    create_ks_query = f"CREATE KEYSPACE {keyspace} WITH REPLICATION = {replication_opts};"
+    create_table_query = (
+        f"CREATE TABLE {keyspace}.{table} (name text PRIMARY KEY, value text) "
+        f"WITH tablets = {{'min_tablet_count': '16'}};"
+    )
+    cql.execute(create_ks_query)
+    cql.execute(create_table_query)
+
+    resource_dir = "test/resource/sstables/fully_partially_contained_ssts"
```

Comment:
How large is this set? In both, bytes and number of rows

Reply:
2503 rows, 5,6 MiB (5.912.682) on disk, 3 SSTables

Reply:
With quite large test I wrote for https://github.com/scylladb/scylladb/pull/26834 where the issue was initially found
just identified the case when the loop was broken prematurely and stole these sstables

Reply:
It's better to generate the sstables on the fly with INSERT and nodetool flush.

You need just two keys per sstable, no? To generate the token range.